git_theta/updates/ia3.py

```diff
@@ -0,0 +1,38 @@
+"""A class for handling activations scaling using ia3 vectors."""
+
+import logging
+from typing import Optional, Any
+from git_theta.updates import IncrementalUpdate
+from git_theta import params
+import numpy as np
+
+Parameter = Any
+
+
+class IA3Update(IncrementalUpdate):
+    """An update where activations are scaled."""
+
+    def __init__(self, serializer: params.Serializer):
+        super().__init__(serializer)
+
+    @property
+    def name(self):
+        return "ia3"
+
+    async def calculate_update(
+        self, parameter: Parameter, previous_parameter: Parameter
+    ) -> Parameter:
+        # since IA3 scales activations, weight.T[1:].shape can be considered as output dimensions and the update scalar vector/ matrix will be of shape = weight[1:].shape
+        # we transpose parameter weights so that output dimensions are in the end
+        parameter = parameter.T
```

Comment:
This assumption about where the output dims are is pytorch specific, a tf implementation has them as the trailing dim by default.

Probably needs an argument about which dims are the broadcast dims? Or it might be easier to have the argument be which dims have ia3 applied.

Also in T5X there are cases where there are multiple ia3 dims (as attention weights are stored as 3d tensors) so we probably need them to be tuples and adjustable per parameter?